design-system/packages/fields/src/DatePicker/index.tsx

```diff
@@ -0,0 +1,79 @@
+/** @jsx jsx */
+
+import React, { Fragment, useCallback } from 'react';
+import FocusLock from 'react-focus-lock';
+import { jsx } from '@keystone-ui/core';
+import { PopoverDialog, usePopover } from '@keystone-ui/popover';
+
+import { Calendar } from './Calendar';
+import { InputButton } from './components/InputButton';
+import { formatDMY, formatDateType } from '../utils/dateFormatters';
+import { DateType } from '../types';
+
+export type DateInputValue = string | undefined;
+
+export type DatePickerProps = {
+  onUpdate: (value: DateType) => void;
+  onClear: () => void;
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
```

Comment:
@mitchellhamilton The onChange callback isn't even being used by the component :/. I've removed it entirely
But out of curiosity what's the reason for wanting to obfuscate the event from the user? I've noticed that a lot with components in our codebase, and its markedly different from the general concensus of my previous team

Reply:
Not sure which components your referring to because in the fields package, the only components that don't do regular onChange events are select and switch which don't use things that emit change events?

In general though, why I like abstracting away events from anything other than a regular input or checkbox is so that we can change how the component works, like a native select to something else or etc. without changing the API.

Reply:
Good to know, thanks for the insight 🎉